jscomp/bsb/bsb_helper_main.ml

```diff
@@ -17,24 +17,101 @@
  * but WITHOUT ANY WARRANTY; without even the implied warranty of
  * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  * GNU Lesser General Public License for more details.
- * 
+ *
  * You should have received a copy of the GNU Lesser General Public License
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
+let normalize cwd s =
+  Ext_filename.normalize_absolute_path (Ext_filename.combine cwd s )
+
+let process_include cwd s =
+  let i = Ext_string.rindex_neg s '?'   in
+  if i < 0 then
+    normalize cwd s
+  else
+    let dir = String.sub s 0 i in
+    normalize cwd dir
+
+let main_module = ref None
+
+let set_main_module modulename =
+    main_module := Some modulename
+
+let includes :  _ list ref = ref []
+
+let add_include cwd dir =
+  includes := process_include cwd dir :: !includes
+
+let batch_files = ref []
+let collect_file name =
+  batch_files := name :: !batch_files
 
 let output_prefix = ref None
 let dev_group = ref 0
-let annoymous other = ()
+let link_flag = ref None
+let pack_flag = ref None
+
+let anonymous filename =
+  collect_file filename
 let usage = "Usage: bsb_helper.exe [options] \nOptions are:"
 let () =
     Arg.parse [
     "-oprefix", Arg.String (fun x -> output_prefix := Some x),
     " Set output prefix for -bs-MD (internal use)";
-    "-g", Arg.Int (fun i -> dev_group := i ), 
+    "-g", Arg.Int (fun i -> dev_group := i ),
     " Set the dev group (default to be 0)"
     ;
-    "-MD", Arg.String (fun x -> Depends_post_process.handle_bin_depfile !output_prefix x !dev_group ),
-    " (internnal)Generate dep file for ninja format(from .ml[i]deps)";
+    "-MD", Arg.String (fun x -> Depends_post_process.handle_bin_depfile ~compilation_kind:Js !output_prefix x !dev_group ),
+    " (internal)Generate dep file for ninja format(from .ml[i]deps)";
+    "-MD-bytecode", Arg.String (fun x -> Depends_post_process.handle_bin_depfile ~compilation_kind:Bytecode !output_prefix x !dev_group ),
+    " (internal)Generate dep file for ninja format(from .ml[i]deps)";
+    "-MD-native", Arg.String (fun x -> Depends_post_process.handle_bin_depfile ~compilation_kind:Native !output_prefix x !dev_group ),
+    " (internal)Generate dep file for ninja format(from .ml[i]deps)";
+
+    (* Used for linking *)
+    "-bs-main", (Arg.String set_main_module),
+    " set the main entry module. Only used in conjunction with -link-bytecode and -link-native";
+
+    "-I",  (Arg.String (add_include (Sys.getcwd ()))),
+    " add dir to search path for the linker and packer";
```

Comment:
why do we need this flag? also `Sys.getcwd` is always on  the code path which is not necessary for js targets